android/src/main/cpp/NativeProxy.cpp

```diff
@@ -584,9 +324,233 @@ int NativeProxy::subscribeForKeyboardEvents(
+#endif
+
+  auto getCurrentTime = bindThis(&NativeProxy::getCurrentTime);
+
+  auto requestRender = bindThis(&NativeProxy::requestRender);
+
+#ifdef RCT_NEW_ARCH_ENABLED
+  auto synchronouslyUpdateUIPropsFunction =
+      bindThis(&NativeProxy::synchronouslyUpdateUIProps);
+#else
+  auto configurePropsFunction = bindThis(&NativeProxy::configureProps);
+#endif
+
+  auto registerSensorFunction = bindThis(&NativeProxy::registerSensor);
+  auto unregisterSensorFunction = bindThis(&NativeProxy::unregisterSensor);
+
+  auto setGestureStateFunction = bindThis(&NativeProxy::setGestureState);
+
+  auto subscribeForKeyboardEventsFunction =
+      bindThis(&NativeProxy::subscribeForKeyboardEvents);
+
+  auto unsubscribeFromKeyboardEventsFunction =
+      bindThis(&NativeProxy::unsubscribeFromKeyboardEvents);
+
+  auto progressLayoutAnimation =
+      bindThis(&NativeProxy::progressLayoutAnimation);
+
+  auto endLayoutAnimation = [this](int tag, bool isCancelled, bool removeView) {
+    this->layoutAnimations_->cthis()->endLayoutAnimation(
+        tag, isCancelled, removeView);
+  };
+
+  return {
+      requestRender,
+#ifdef RCT_NEW_ARCH_ENABLED
+      synchronouslyUpdateUIPropsFunction,
+#else
+      updatePropsFunction,
+      scrollToFunction,
+      measureFunction,
+      configurePropsFunction,
+#endif
+      getCurrentTime,
+      progressLayoutAnimation,
+      endLayoutAnimation,
+      registerSensorFunction,
+      unregisterSensorFunction,
+      setGestureStateFunction,
+      subscribeForKeyboardEventsFunction,
+      unsubscribeFromKeyboardEventsFunction,
+  };
+}
+
+void NativeProxy::setGlobalProperties(
+    jsi::Runtime &jsRuntime,
+    const std::shared_ptr<jsi::Runtime> &reanimatedRuntime) {
+  auto workletRuntimeValue =
+      jsRuntime.global()
+          .getPropertyAsObject(jsRuntime, "ArrayBuffer")
+          .asFunction(jsRuntime)
+          .callAsConstructor(jsRuntime, {static_cast<double>(sizeof(void *))});
+  uintptr_t *workletRuntimeData = reinterpret_cast<uintptr_t *>(
+      workletRuntimeValue.getObject(jsRuntime).getArrayBuffer(jsRuntime).data(
+          jsRuntime));
+  workletRuntimeData[0] = reinterpret_cast<uintptr_t>(reanimatedRuntime.get());
+
+  jsRuntime.global().setProperty(
+      jsRuntime, "_WORKLET_RUNTIME", workletRuntimeValue);
+
+#ifdef RCT_NEW_ARCH_ENABLED
+  jsRuntime.global().setProperty(jsRuntime, "_IS_FABRIC", true);
+#else
+  jsRuntime.global().setProperty(jsRuntime, "_IS_FABRIC", false);
+#endif
+
+  auto version = getReanimatedVersionString(jsRuntime);
+  jsRuntime.global().setProperty(jsRuntime, "_REANIMATED_VERSION_CPP", version);
+}
+
+void NativeProxy::setupLayoutAnimations() {
+  auto weakModule =
+      std::weak_ptr<NativeReanimatedModule>(nativeReanimatedModule_);
+
+  layoutAnimations_->cthis()->setAnimationStartingBlock(
+      [weakModule](
+          int tag, int type, alias_ref<JMap<jstring, jstring>> values) {
+        auto reaModule = weakModule.lock();
```

Comment:
I know that `module` will be a keyword in C++20 but it's also used in various many places so for now let's be consistent and use `module` here and some time in the future we will rename it everywhere.
```suggestion
        auto module = weakModule.lock();
```

Reply:
I'm pretty sure C++20 shouldn't break `module` being used as an identifier anyway, but let's not get into that. 😛